tester/cs/testing-with-gitlab-ci.texy

```diff
@@ -0,0 +1,124 @@
+
+V dnešní době se hojně používá GitLab, důvodů je více - intuitivní ovládání GUI, ale hlavní také je výhoda je, že GitLab včetně GitLab CI lze provzovat na vlastním serveru a tím je zaručeno soukromí, protože je opensource. Ale také běž na https://gitlab.com/ .
+
+Pojďme si ukáázat, jak jednodušše přidat CI na GL.
+
+Jak funguje GitLab CI
+=====================
+
+GitLab CI((Continuous Integration)) (dále jen CI) je velmi populární služba pro automatizaci [průběžné integrace](https://cs.wikipedia.org/wiki/Průběžná_integrace). Je poskytována zdarma((Pro soukromé repozitáře má omené CI minuty, neýká se instnce CI na vlastním serveru.)) a nabízí automatizované spouštění testů, deploy aplikací atd. po zápisu změny do repozitáře. Je plně integrována do prostředí GitHubu.
+
+CI se konfiguruje souborem `.gitlab-ci.yml` umístěným v kořenovém adresáři projektu. Obsahuje několik bloků, které si podrobněji popíšeme:
+
+Je potřeba nedefinovat stages, aby CI vědělo, jak se mají úlohy po sobě spouštět. Úlohy se spoští paralelně v jednotlivé stagi.
+
+Každý job by měl mít definován nějaký image, jinak používá image defaultní, definovaný na CI runneru. Lze ovšem definovat i globálně pro všechny joby v rootu konfigurace.
+
+Proměnné prostředí
+------------------
+
+V sekci `variables` nastavujeme další proměnnépro pro prostředí pro běh testů. Pro každou odrážku se skripty provedou samostatně a v jedné odrážce lze definovat více proměnných. Dále v textu budeme používat proměnnou `TESTER_PHP_BIN`, abychom mohli testy pouštět také s parametrem `-p php-cgi`.
+
+```yaml
+variables:
+  - TESTER_PHP_BIN="php"
+  - TESTER_PHP_BIN="php-cgi"
+```
+
+Spouštění testů
+---------------
+
+Testy se spouštějí v sekci `script`, kde stačí zavolat tester. Předpokládejme, že testy jsou umístěny ve složce `tests/` a používáte vlastní `php.ini` umístěné ve složce s testy. Pomocí `-s` si necháme vypsat informace o přeskočených testech. Jako binárku nastavíme v parametru `-p` hodnotu proměnné `TESTER_PHP_BIN`, kterou jsme si nastavili v bloku `env`.
+
+```yaml
+script:
+  - ./vendor/bin/tester -p $TESTER_PHP_BIN -s -c ./tests/php.ini ./tests
+```
+
+Nastavení dalších služeb
+------------------------
+
+CI nám umožňuje zapnout další služby. Například Redis server na který ukládáte cache.
+
+```yaml
+services:
+  - redis-server:lastest
+```
+
+Inicializace databáze:
+----------------------
+
+Databáze MySQL je již předinstalována. Běží na adrese `127.0.0.1` a přihlásit se můžete na účet `travis` nebo `root` bez hesla. Pokud byste chtěli spouštět integrační testy, můžete si do ni naimportovat testovací databázi, kterou máte například v souboru `tests/testbase.sql`:
+
+```yaml
+services:
+    - mysql:latest
+
+before_script:
+  - mysql -u root -e 'CREATE DATABASE testbase;'
+  - mysql -u root testbase < tests/testbase.sql
+```
+
+Výsledná ukázka konfiguračního souboru
+--------------------------------------
+
+Celý `.gitlab-ci.yml` pak vypadá takto:
+
+```yml
```

Comment:
Osobne mi prijde tenhle priklad slozitej. Na prvni pokus bych zkusil uzivatele seznamit s minimalni konfiguraci, jak spustit nette tester. Ne kazdej potrebuje mysql k testovani, skoro bych rekl, ze se i lidi boji pouzivat nejake dalsi sluzby pri testovani.

Reply:
máš nápad, jak to napsatM